google/cloud/internal/credentials.h

```diff
@@ -47,6 +48,18 @@ std::shared_ptr<Credentials> MakeAccessTokenCredentials(
     std::string const& access_token,
     std::chrono::system_clock::time_point expiration);
 
+struct DelegatesOption {
```

Comment:
Q: Is the plan that all of the user-visible options for the new credentials library will exist in this single `credentials.h` file? Or will they be spread around in a few files? I ask because it would be nice for them to exist in a single file. This file (`credentials.h`) is fine with me. Otherwise, consider a common location like `credentials_options.h` or similar.

Reply:
I was planning to keep them all in `credentials.h`, we will see, I expect maybe another handful of options.